server/pulp/server/db/manage.py

```diff
@@ -115,16 +115,34 @@ def ensure_database_indexes():
     MongoEngine based models require a manual check as they will only create indexes if the
     collection does not already exist.
     """
-    model.RepositoryContentUnit.ensure_indexes()
-    model.Repository.ensure_indexes()
-    model.ReservedResource.ensure_indexes()
-    model.TaskStatus.ensure_indexes()
-    model.Worker.ensure_indexes()
+
+    _ensure_index(model.RepositoryContentUnit)
+    _ensure_index(model.Repository)
+    _ensure_index(model.ReservedResource)
+    _ensure_index(model.TaskStatus)
+    _ensure_index(model.Worker)
 
     # Load all the model classes that the server knows about and ensure their inexes as well
     plugin_manager = PluginManager()
     for model_class in plugin_manager.unit_models.itervalues():
+        _ensure_index(model_class)
+
+
+def _ensure_index(model_class):
```

Comment:
Later versions of mongoengine have both `ensure_index` and `ensure_indexes` methods on Documents. It may be better to rename this to `_ensure_indexes` just to make it consistent with the other methods it calls.